### PR TITLE
Blog and article page

### PR DESCRIPTION
**Pull request by Shembou**

### CODE REVIEW

File: next/src/components/_article/hero/hero.module.scss [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    display: grid;
+    gap: 14px;
+
+    .categories {
+      display: flex;
+      flex-wrap: wrap;
+      gap: clamp(10px, calc(16vw/7.68), 24px);
+
+      .heading {
+        &:before {
```
Comment (KryptonumDev) [inline next/src/components/_article/hero/hero.module.scss:19]: Dobrą praktyką jest używanie :: przed pseudo elementami zamiast :

File: next/src/components/_global/contentComponent/contentSection.tsx
[the remark this answers is not in the document]
Comment (Shembou) [reply next/src/components/_global/contentComponent/contentSection.tsx:1]: Hmm, zobaczę co da się zrobić. Jeżeli znajdę czas spróbuję to przerobić. W sumie w tym przypadku nie występuje raczej CLS, ale to będzie i tak 1 kliencki komponent mniej

File: sanity/sanity.config.js [hunk 2]
```diff
@@ -35,28 +31,40 @@ export default defineConfig({
     deskTool({
       structure: (S) =>
         S.list()
-          .title("Pages")
+          .title('Pages')
```
Comment (milewskibogumil) [inline sanity/sanity.config.js:35]: Change the name to 'Podstrony' - it's for a Polish client

File: sanity/schemas/collectionTypes/blog/categories.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+export default {
+  name: 'blog_categories',
+	title: 'Kategorie',
```
Comment (milewskibogumil) [inline sanity/schemas/collectionTypes/blog/categories.js:3]: Change the title for 'Kategorie bloga'

File: sanity/schemas/components/ctaWithBackgroundImage.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      description: 'Relatywny albo absolutny link (https://)',
+      validation: (Rule) =>
+        Rule.custom((value) => {
+          if (
+            value &&
+            !value.startsWith('/') &&
+            !value.startsWith('https://') &&
+            !value.startsWith('#')
+          ) {
+            return 'Incorrect URL.'
```
Comment (milewskibogumil) [inline sanity/schemas/components/ctaWithBackgroundImage.js:59]: Change the language of the error to Polish language

File: sanity/schemas/components/grid2Images.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    {
+      title: 'Zdjęcia',
+      name: 'images',
+      type: 'array',
+      of: [{type: 'image'}],
+    },
+    {
+      title: 'Źródło',
+      name: 'source',
+      type: 'markdown',
```
Comment (milewskibogumil) [inline sanity/schemas/components/grid2Images.js:15]: - 'Źródło' is incomprehensible.
- This field is for URL - why type of that is markdown. Use `[URL](https://www.sanity.io/docs/url-type)` type.

File: sanity/schemas/components/imageSource.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  fields: [
+    {
+      title: 'Zdjęcie',
+      name: 'image',
+      type: 'image',
+    },
+    {
+      title: 'Źródło',
+      name: 'source',
+      type: 'markdown',
```
Comment (milewskibogumil) [inline sanity/schemas/components/imageSource.js:14]: Change the type and title globally for that field

File: sanity/schemas/components/orderedListArray.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+      ]
+    }
+  ],
+  preview: {
+    select: {
+      title: 'array',
+    },
+    prepare({ title }) {
+      return {
+        title: `[Lista numerowana] - ${title.length} elementy`,
```
Comment (milewskibogumil) [inline sanity/schemas/components/orderedListArray.js:23]: type, should be: elementów*

File: sanity/schemas/components/quote.js [hunk 8]
```diff
@@ -19,10 +19,13 @@ export default {
...
       title: 'heading',
+      subtitle: 'quote',
     },
-    prepare({title}) {
+    prepare({title, subtitle}) {
       return {
-        title: `${removeMarkdown(title)}`,
+        title: title
+          ? `[Cytat] - ${removeMarkdown(title)}`
+          : `[Cytat] - ${removeMarkdown(subtitle)}`,
```
Comment (milewskibogumil) [inline sanity/schemas/components/quote.js:28]: Isn't it be more clear to write it like: `title: `[Cytat] - ${removeMarkdown(title || subtitle)}``

File: next/src/app/blog/[slug]/page.tsx [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+    slug: slug.current,
+  }));
+}
+
+export async function generateMetadata({ params: { slug } }: { params: { slug: string } }) {
+  const { seo } = await getMetadata(slug);
+  return SEO({
+    title: seo?.title,
+    description: seo?.description,
+    url: '/co-robie',
```
Comment (milewskibogumil) [inline next/src/app/blog/[slug]/page.tsx:21]: Wrong path for SEO provided. Take a look globally for SEO.

File: next/src/app/blog/[slug]/page.tsx [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+  });
+  return blogPosts;
+}
+
+async function getMetadata(slug: string) {
+  const { page } = await sanityFetch<BlogPostPage>({
+    query: /* groq */ `{
+      "page": *[_type=="blog_entries" && slug.current==$slug][0] {
+      seo {
+       ...,
```
Comment (milewskibogumil) [inline next/src/app/blog/[slug]/page.tsx:54]: Don't use merge operator anywhere in query. Explicitly declare them. See: https://www.sanity.io/docs/high-performance-groq#9b1f0b9a4b19. Globally change that.

File: next/src/app/blog/[slug]/page.tsx [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+                dimensions {
+                  height,
+                  width
+                }
+              }
+          }
+        },
+        hero_Title,
+        content[] {
+          ...
```
Comment (milewskibogumil) [inline next/src/app/blog/[slug]/page.tsx:88]: Don't use merge operator anywhere in query. Explicitly declare them. See: https://www.sanity.io/docs/high-performance-groq#9b1f0b9a4b19. Globally change that.

File: next/src/components/_article/hero/hero.module.scss (hunk 1 above)
Comment (milewskibogumil) [reply next/src/components/_article/hero/hero.module.scss:19]: I agree. Look: https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-elements

File: next/src/components/_global/ctaWithBackgroundImage/ctaWithBackgroundImage.tsx [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+export default function CtaWithBackgroundImage({ data: { cta, title, subtitle } }: Props) {
+  return (
+    <div className={styles.ctaWithBackgroundImage}>
+      <Markdown.h2 className={styles.title}>{title}</Markdown.h2>
+      {subtitle && <Markdown className={styles.subtitle}>{subtitle}</Markdown>}
+      {cta.text && (
+        <Button
+          data={cta}
+          svg={false}
+          theme='secondary'
```
Comment (milewskibogumil) [inline next/src/components/_global/ctaWithBackgroundImage/ctaWithBackgroundImage.tsx]: You doesn't declare theme from Sanity, passing the `cta` object?

File: next/src/components/_global/portableContent/portableContent.module.scss [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+
+  color: var(--dark-neutral-900);
+
+  .link {
+    color: var(--coffee-500);
+  }
+
+  .quote {
+    margin-top: 35px;
+    margin-bottom: 35px;
```
Comment (milewskibogumil) [inline next/src/components/_global/portableContent/portableContent.module.scss:13]: Shorthand is `margin: 35px 0;`

File: next/src/components/_global/portableContent/portableContent.tsx [hunk 14]
```diff
@@ -0,0 +1,153 @@
...
+    number: ({ children }) => <ol className={styles.portableList}>{children}</ol>,
+  },
+  marks: {
+    link: ({ value, children }) => {
+      const { href }: { href: string } = value as { href: string };
+      return (
+        <a
+          href={href}
+          target='_blank'
+          rel='noreferrer'
```
Comment (milewskibogumil) [inline next/src/components/_global/portableContent/portableContent.tsx:134]: Should be added `rel=noopener`

**Review by milewskibogumil — CHANGES_REQUESTED** [review]

**Review by milewskibogumil — CHANGES_REQUESTED** [review]

File: next/src/components/_blog/blogReferenceSection/blogReferenceSection.tsx [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+                type='h1'
+                key={i}
+              >
+                {category.name}
+              </Heading>
+            ))}
+        </div>
+        <Markdown.h2 className={styles.title}>{hero_Title}</Markdown.h2>
+        <Markdown>{hero_Description}</Markdown>
+        <ReadingTime text={JSON.stringify(content, null, 2)}/>
```
[the remark this answers is not in the document]
Comment (Shembou) [reply next/src/components/_blog/blogReferenceSection/blogReferenceSection.tsx:47]: to make it work. There is no _contentRaw, or any other similar field in groq. Therefore I stringify it so it works properly

File: next/src/app/blog/kategoria/[category]/page.tsx [hunk 16]
```diff
@@ -0,0 +1,224 @@
...
+  return page;
+}
+
+async function getBlogCategories() {
+  const { blogCategories } = await sanityFetch<BlogPage>({
+    query: /* groq */ `{
+      "blogCategories": *[_type=="blog_categories"] {
+        slug {
+          current
+        }
```
[the remark this answers is not in the document]
Comment (Shembou) [reply next/src/app/blog/kategoria/[category]/page.tsx:88]: good to know

File: next/src/components/_global/portableContent/portableContent.module.scss (hunk 13 above)
Comment (Shembou) [reply next/src/components/_global/portableContent/portableContent.module.scss:13]: good to know

File: next/src/components/_article/hero/hero.module.scss (hunk 1 above)
Comment (Shembou) [reply next/src/components/_article/hero/hero.module.scss:19]: good to know

File: sanity/schemas/components/imageSource.js (hunk 6 above)
Comment (Shembou) [reply sanity/schemas/components/imageSource.js:14]: same as above

File: next/src/app/blog/[slug]/page.tsx (hunk 9 above)
Comment (Shembou) [reply next/src/app/blog/[slug]/page.tsx:21]: rest will be set in dev branch

File: next/src/app/blog/[slug]/page.tsx (hunk 10 above)
Comment (Shembou) [reply next/src/app/blog/[slug]/page.tsx:54]: global changes on dev branch

File: next/src/components/_global/ctaWithBackgroundImage/ctaWithBackgroundImage.tsx (hunk 12 above)
Comment (Shembou) [reply next/src/components/_global/ctaWithBackgroundImage/ctaWithBackgroundImage.tsx]: This cta is unique. User has no access to set theme in this button. The theme is set to reduce the amount of styles.

File: next/src/components/_global/portableContent/portableContent.tsx (hunk 14 above)
Comment (Shembou) [reply next/src/components/_global/portableContent/portableContent.tsx:134]: what's the difference?

File: next/src/app/blog/kategoria/[category]/[number]/page.tsx [hunk 17]
```diff
@@ -0,0 +1,275 @@
...
+async function getMetadata() {
+  const { page } = await sanityFetch<BlogPage>({
+    query: /* groq */ `{
+      "page": *[_id=="blogPage"][0] {
+        seo {
+          ...,
+          og_Img {
+            asset -> {
+              url
+            }
```
[the remark this answers is not in the document]
Comment (Shembou) [reply next/src/app/blog/kategoria/[category]/[number]/page.tsx:137]: don't instead of doesn't 😅 
are you instead of you are

**Pull request merged**